File: complete-binary-tree/complete-binary-tree.py
class TreeNode:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

class CompleteBinaryTree:
    def __init__(self):
        self.root = None

    def insert(self, data):
        if self.root is None:
            self.root = TreeNode(data)
            return

        queue = [self.root]
        while queue:
            current = queue.pop(0)

            if current.left is None:
                current.left = TreeNode(data)
                break
            else:
                queue.append(current.left)

            if current.right is None:
                current.right = TreeNode(data)
                break
            else:
                queue.append(current.right)

    def inorder_traversal(self, node):
        if node is None:
            return
        self.inorder_traversal(node.left)
        print(node.data, end=" ")
        self.inorder_traversal(node.right)

if __name__ == "__main__":
    tree = CompleteBinaryTree()
    tree.insert(10)
    tree.insert(20)
    tree.insert(30)
    tree.insert(40)
    tree.insert(50)

    print("Inorder Traversal of the Complete Tree: ", end="")
    tree.inorder_traversal(tree.root)
    print()

File: complete-binary-tree/complete-binary-tree.java
import java.util.LinkedList;
import java.util.Queue;

class TreeNode {
    int data;
    TreeNode left, right;

    TreeNode(int data) {
        this.data = data;
        this.left = null;
        this.right = null;
    }
}

public class CompleteBinaryTree {
    TreeNode root;

    public void insert(int data) {
        if (root == null) {
            root = new TreeNode(data);
            return;
        }

        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        while (!queue.isEmpty()) {
            TreeNode current = queue.poll();

            if (current.left == null) {
                current.left = new TreeNode(data);
                break;
            } else {
                queue.add(current.left);
            }

            if (current.right == null) {
                current.right = new TreeNode(data);
                break;
            } else {
                queue.add(current.right);
            }
        }
    }

    public void inorderTraversal(TreeNode node) {
        if (node == null) return;
        inorderTraversal(node.left);
        System.out.print(node.data + " ");
        inorderTraversal(node.right);
    }

    public static void main(String[] args) {
        CompleteBinaryTree tree = new CompleteBinaryTree();

        tree.insert(10);
        tree.insert(20);
        tree.insert(30);
        tree.insert(40);
        tree.insert(50);

        System.out.print("Inorder Traversal of the Complete Tree: ");
        tree.inorderTraversal(tree.root);
        System.out.println();
    }
}
